shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala

```diff
@@ -493,6 +502,76 @@ class PickleReader {
 
   private def noSuchConstantTag(tag: Int, len: Int): Nothing =
     errorBadSignature("bad constant tag: " + tag)
+
+  /** Convert
+    *   tp { type name = sym } forSome { sym >: L <: H }
+    * to
+    *   tp { name >: L <: H }
+    * and
+    *   tp { name: sym } forSome { sym <: T with Singleton }
+    * to
+    *   tp { name: T }
+    */
+  private def elimExistentials(boundSyms: List[Symbol], tp: Type)(using Context): Type =
+    // Need to be careful not to run into cyclic references here (observed when
+    // compiling t247.scala). That's why we avoid taking `symbol` of a TypeRef
+    // unless names match up.
+    val isBound = { (tp: Type) =>
+      def refersTo(tp: Type, sym: Symbol): Boolean = tp match {
+        case tp: TypeRef => tp.isLocalRef(sym)
+        /*case tp: TypeVar => refersTo(tp.underlying, sym)
+        case tp : LazyRef => refersTo(tp.ref, sym)*/
+        case _ => false
+      }
+      boundSyms.exists(refersTo(tp, _))
+    }
+    // Cannot use standard `existsPart` method because it calls `lookupRefined`
+    // which can cause CyclicReference errors.
+    /*val isBoundAccumulator = new ExistsAccumulator(isBound, StopAt.Static, forceLazy = true):
+      override def foldOver(x: Boolean, tp: Type): Boolean = tp match
+        case tp: TypeRef => applyToPrefix(x, tp)
+        case _ => super.foldOver(x, tp)*/
+
+    /*def removeSingleton(tp: Type): Type =
+      if (tp isRef defn.SingletonClass) defn.AnyType else tp*/
+    def mapArg(arg: Type) = arg match {
+      case arg: TypeRef if isBound(arg) => arg.symbol.declaredType
+      case _                            => arg
+    }
+    def elim(tp: Type): Type = tp match {
+      /*case tp @ RefinedType(parent, name, rinfo) =>
+        val parent1 = elim(tp.parent)
+        rinfo match {
+          case TypeAlias(info: TypeRef) if isBound(info) =>
+            RefinedType(parent1, name, info.symbol.info)
+          case info: TypeRef if isBound(info) =>
+            val info1 = info.symbol.info
+            assert(info1.derivesFrom(defn.SingletonClass))
+            RefinedType(parent1, name, info1.mapReduceAnd(removeSingleton)(_ & _))
+          case info =>
+            tp.derivedRefinedType(parent1, name, info)
+        }*/
+      case tp @ AppliedType(tycon, args) =>
+        val tycon1 = tycon // tycon.safeDealias
+        if (tycon1 ne tycon) elim(tycon1.appliedTo(args))
```

Comment:
How can `tycon1 ne tycon`?

Reply:
When we actually do the `tycon.safeDealias` which is currently commented out. I'll comment this branch as well; it's confusing.